packages/compass-components/src/components/guide-cue/guide-cue.tsx

```diff
@@ -0,0 +1,222 @@
+import React, {
+  useEffect,
+  useCallback,
+  useRef,
+  useMemo,
+  useState,
+} from 'react';
+import { guideCueService, type ShowCueEventDetail } from './guide-cue-service';
+import { GuideCue as LGGuideCue } from '@leafygreen-ui/guide-cue';
+import type { GroupName } from './guide-cue-groups';
+import { css, cx } from '../..';
+import { rafraf } from '../../utils/rafraf';
+
+const hiddenPopoverStyles = css({
+  display: 'none !important',
+  opacity: '0 !important',
+  transition: 'none !important',
+});
+
+const getDataCueId = ({
+  cueId,
+  groupId,
+}: {
+  cueId: string;
+  groupId?: string;
+}): string => {
+  return `guide-cue-${cueId}${groupId ? `-${groupId}` : ''}`;
+};
+
+type LGGuideCueProps = React.ComponentProps<typeof LGGuideCue>;
+
+type GroupAndStep =
+  | {
+      groupId: GroupName;
+      step: number;
+    }
+  | {
+      groupId?: never;
+      step?: never;
+    };
+
+// omit the props we are handling
+export type GuideCueProps<T> = Omit<
+  LGGuideCueProps,
+  | 'currentStep'
+  | 'refEl'
+  | 'numberOfSteps'
+  | 'onDismiss'
+  | 'open'
+  | 'onPrimaryButtonClick'
+  | 'setOpen'
+> &
+  GroupAndStep & {
+    cueId: string;
+    trigger: ({ ref }: { ref: React.Ref<T> }) => React.ReactElement;
+  };
+
+export const GuideCue = <T extends HTMLElement>({
+  children,
+  trigger,
+  cueId,
+  groupId,
+  step,
+  ...restOfCueProps
+}: React.PropsWithChildren<GuideCueProps<T>>) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const intersectionRef = useRef(true);
+  const refEl = useRef<T>(null);
+  const [readyToRender, setReadyToRender] = useState(false);
+
+  const cueData = useMemo(() => {
+    if (!groupId) {
+      return { cueId, step: 1 };
+    }
+    return { cueId, groupId, step };
+  }, [cueId, groupId, step]);
+
+  const observerCallback = useCallback(
+    (entries: IntersectionObserverEntry[]) => {
+      const [entry] = entries;
+      intersectionRef.current = entry.isIntersecting;
```

Comment:
I might be missing something here, but the way this intersectionRef is used in other hooks, you probably want it to be part of the component state, not just a ref? You use it for rendering, but nothing guarantees that the render will be triggered if this value is just a ref

Reply:
I changed it to be a component state in [8485e90](https://github.com/mongodb-js/compass/pull/4524/commits/8485e90021c490303e0e3e18779464015779c8aa). I had added it prior to adding dismissible behaviour on GuideCue